reactor-netty-http/src/main/java/reactor/netty/http/server/HttpRouteHandler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.netty.http.server;
+
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.reactivestreams.Publisher;
+import reactor.util.annotation.Nullable;
+
+/**
+ * @author Stephane Maldini
+ * @author chentong
+ */
+public class HttpRouteHandler
```

Comment:
Basically from `HttpRouteHandler` we need only the `path`. Can we expose an interface that will have access only to `path`?

Reply:
expose an interface `RouteHandlerMetadata `
~~~
public interface RouteHandlerMetadata {
	
	String getRoutePath();
}
~~~

use `RouteHandlerMetadata`  to access `HttpRouteHandler's` metada like `path`?

Reply:
good idea @violetagg, yeah an interface for metadata sounds good @ctlove0523. This will make it extendable. Of course, the `HttpServerRoute#comparator` API will need to be based on said interface.

Reply:
the interface would be a good spot to offer two basic comparators: lexical and lexical-with-template

Reply:
For this one you mean something similar to what we have with `reactor.netty.http.server.logging.AccessLogFactory` right?

Reply:
(by `lexical-with-template` I mean one that prioritize joker characters differently, like in the original issue)

Reply:
@simonbasle @violetagg let ou'rnbasle @violetagg let our users to define the order of route(we need to define new api)？
If one user add routes in the below order：
Route1(order = 1)
Route2(order = DEFAULT_ORDER),
Route3(order = 4)
Route4(order =DEFAULT_ORDER）
Route5(order = 2)

when get a request use `Route1 -> Route5 -> Route3 -> Route2 -> Route4` order to match request.

Reply:
@ctlove0523 The last proposal seems similar to the use case where we have the current implementation and you order the routes by yourself while configuring them.

Reply:
I don't think asking the users to provide an index / priority is good ux, at least not terribly better than just asking them to invoke the routes in the desired order.

We _might_ want to consider internally adding the index information (as in "in which order did the `route` method did get invoked", eg. by incrementing an `AtomicInteger` and passing that to the constructor) so that users can restore "insertion order" so to speak, through a third built-in `Comparator`.

The index information doesn't even need to be public if the `Comparator` is implemented as package-private inside `HttpRouteHandler` and exposed as a static factory method in the `RouteHandlerMetadata` interface.

wdyt @violetagg @ctlove0523 ?

Reply:
@violetagg use call method order is unstable,user explicit define route order is better. I will make new PR refer to your and simonbasle‘s review.

Reply:
(please continue in this PR if possible, to limit leaving a breadcrumb trail of discussions in several closed PRs)